x-pack/plugins/infra/public/observability_logs/log_stream_query_state/src/search_bar_state_service.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FilterManager, QueryStringContract } from '@kbn/data-plugin/public';
+import { map } from 'rxjs/operators';
+import type { InvokeCreator } from 'xstate';
+import type { LogStreamQueryContext, LogStreamQueryEvent } from './types';
+
+export const subscribeToQuerySearchBarChanges =
+  ({
+    queryStringService,
+  }: {
+    queryStringService: QueryStringContract;
+  }): InvokeCreator<LogStreamQueryContext, LogStreamQueryEvent> =>
+  (context) =>
+    queryStringService.getUpdates$().pipe(
+      map(() => queryStringService.getQuery()),
+      map((query): LogStreamQueryEvent => {
+        return {
+          type: 'QUERY_FROM_SEARCH_BAR_CHANGED',
+          query,
+        };
+      })
+    );
+
+export const updateQueryInSearchBar =
+  ({ queryStringService }: { queryStringService: QueryStringContract }) =>
+  (context: LogStreamQueryContext, event: LogStreamQueryEvent) => {
+    if (!('query' in context)) {
+      return;
+    }
+
+    queryStringService.setQuery(context.query);
```

Comment:
**nit**: I stopped for a second to read the condition, IMO it may be more immediate to understand something without the guard in this case:
```js
if ('query' in context) {
  queryStringService.setQuery(context.query);
}
```

Reply:
Same for the following `updateFiltersInSearchBar` function

Reply:
hm, yes. since the guarded operation is a one-liner I could guard it directly instead of expressing it as an early return. :+1: